spec/spec_helper_acceptance_local.rb

```diff
@@ -0,0 +1,402 @@
+require 'master_manipulator'
+require 'singleton'
+
+class Helper
+  include PuppetLitmus
+  include BoltSpec
+  include Singleton
+  def inventory_hash
+    @inventory_hash ||= inventory_hash_from_inventory_file
+  end
+
+  def target_roles(roles)
+    # rubocop:disable Style/MultilineBlockChain
+    inventory_hash['groups'].map { |group|
+      group['targets'].map { |node|
+        { name: node['uri'], role: node['vars']['role'] } if node['vars']['role'].include? roles
+      }.reject { |val| val.nil? }
+    }.flatten
+    # rubocop:enable Style/MultilineBlockChain
+  end
+end
+
+class LitmusAgentRunner
+  include MasterManipulator::Site
+  include PuppetLitmus
+
+  def create_site_pp_host(master_host, opts = {})
+    opts[:manifest] ||= ''
+    opts[:node_def_name] ||= 'default'
+    ENV['TARGET_HOST'] = master_host
+    master_certname = Helper.instance.run_shell('puppet config print certname').stdout.rstrip
+
+    default_def = <<-MANIFEST
+node default {
+}
+MANIFEST
+
+    node_def = <<-MANIFEST
+node #{opts[:node_def_name]} {
+#{opts[:manifest]}
+}
+MANIFEST
+
+    if opts[:node_def_name] != 'default'
+      node_def = "#{default_def}\n#{node_def}"
+    end
+
+    site_pp = <<-MANIFEST
+filebucket { 'main':
+server => '#{master_certname}',
+path   => false,
+}
+
+File { backup => 'main' }
+
+#{node_def}
+MANIFEST
+
+    site_pp
+  end
+
+  def generate_site_pp(agents_hash)
+    master = WebSphereHelper.host_by_role('master')
+    ENV['TARGET_HOST'] = master.first
+    # Initialize a blank site.pp
+    site_pp = create_site_pp_host(master.first, manifest: '')
+    agents_hash.each do |agent, manifest|
+      # pull out the node specific block for the site.pp
+      node_block = create_site_pp_host(master.first, manifest: manifest, node_def_name: agent)
+      node_block = node_block.split("node #{agent}")[-1]
+      node_block = "node '#{agent}'#{node_block}"
+      site_pp << node_block
+    end
+    site_pp
+  end
+
+  def copy_site_pp(site_pp, _opts = {})
+    master = WebSphereHelper.host_by_role('master')
+    ENV['TARGET_HOST'] = master.first
+    environment_base_path = run_shell('puppet config print environmentpath').stdout.rstrip
+    prod_env_site_pp_path = File.join(environment_base_path, 'production', 'manifests', 'site.pp')
+    site_pp_dir = File.dirname(prod_env_site_pp_path)
+    site_pp_file = File.new('site.pp', 'w')
+    site_pp_file.write(site_pp)
+    site_pp_file.close
+    bolt_upload_file('site.pp', prod_env_site_pp_path)
+    run_shell("chmod 0744 #{site_pp_dir}")
+    run_shell("chmod 0744 #{prod_env_site_pp_path}")
+    run_shell("chown -R pe-puppet:pe-puppet #{site_pp_dir}")
+    run_shell("chown -R pe-puppet:pe-puppet #{prod_env_site_pp_path}")
+    print "site.pp:\n#{site_pp}"
+  end
+
+  def execute_agent_on(host, manifest = nil, opts = {})
+    if manifest
+      site_pp = generate_site_pp(host => manifest)
+      copy_site_pp(site_pp, opts)
+    end
+    cmd =  'agent --test --environment production'
```

Comment:
This [was previously an Array](https://github.com/puppetlabs/puppetlabs-websphere_application_server/pull/197/files#diff-174f7dfb508155027cb4c3e9a1840591L111), so I think you'll need spaces in the subsequent appends to ensure the cmd works: